packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -1381,6 +1382,42 @@ class Transform extends SingleChildRenderObjectWidget {
         assert(scale == null || (scaleX == null && scaleY == null), "If 'scale' is non-null then 'scaleX' and 'scaleY' must be left null"),
         transform = Matrix4.diagonal3Values(scale ?? scaleX ?? 1.0, scale ?? scaleY ?? 1.0, 1.0);
 
+  // Computes a rotation matrix for an angle in radians, attempting to keep rotations
+  // at integral values for angles of 0, π/2, π, 3π/2.
+  static Matrix4 _computeRotation(double radians) {
+    if (radians == 0.0) {
+      return Matrix4.identity();
+    }
+    final double sin = math.sin(radians);
+    if (sin == 1.0) {
+      return _createZRotation(1.0, 0.0);
+    }
+    if (sin == - 1.0) {
+      return _createZRotation(-1.0, 0.0);
+    }
+    if (math.cos(radians) == -1.0) {
+      return _createZRotation(0.0, -1.0);
+    }
+    return Matrix4.rotationZ(radians);
```

Comment:
Since you've already computed both sin/cos, use the _createZRotation method anyway. It works fine even if the values aren't exactly 0/1.

Reply:
And then you avoid computing sin/cos twice (once here for fast path and once again in the Matrix4 method).